shared/test/test_animations.rb

```diff
@@ -377,6 +377,50 @@ def test_doesnt_mask_unrelated_errors
     soft_delete(filename)
   end
 
+  def test_get_object_with_latest_version
+    filename = @api.randomize_filename('test.png')
+    delete_all_animation_versions(filename)
+
+    # Create an animation file
+    upload(filename, 'stub-v1-body')
+    latest_version_id = upload(filename, 'stub-v2-body')
+
+    # Delete it.
+    soft_delete(filename)
+
+    # Attempt to get object with special key to get latestVersion
+    response = AnimationBucket.new.get(@channel_id, filename, nil, 'latestVersion')
+    assert_equal response[:version_id], latest_version_id
+
+    soft_delete(filename)
+  end
+
+  def test_get_object_with_latest_version_of_non_deleted
+    filename = @api.randomize_filename('test.png')
+    delete_all_animation_versions(filename)
+
+    # Create an animation file
+    upload(filename, 'stub-v1-body')
+    latest_version_id = upload(filename, 'stub-v2-body')
+
+    # Attempt to get object with special key to get latestVersion
+    response = AnimationBucket.new.get(@channel_id, filename, nil, 'latestVersion')
+    assert_equal response[:version_id], latest_version_id
+
+    soft_delete(filename)
+  end
+
+  def test_get_object_with_latest_version_for_missing_animation
+    filename = @api.randomize_filename('test.png')
+    delete_all_animation_versions(filename)
+
+    # Attempt to get object with special key to get latestVersion
+    response = AnimationBucket.new.get(@channel_id, filename, nil, 'latestVersion')
+    assert_equal response[:status], 'NOT_FOUND'
+
+    soft_delete(filename)
```

Comment:
This one's also redundant - and I almost expect it to fail.  Anyway, I think it can be removed.